src/main/java/com/vmware/vipclient/i18n/messages/service/CacheService.java

```diff
@@ -6,127 +6,101 @@
-        Cache c = VIPCfg.getInstance().getCacheManager().getCache(VIPCfg.CACHE_L3);
-        if (c != null) {
-            f = c.get(cacheKey) != null;
-        }
-        return f;
+        cacheItem.setCacheItem(itemToCache);
     }
 
     public boolean isContainStatus() {
-        boolean f = false;
         String cacheKey = dto.getTransStatusAsCacheKey();
-        Cache c = VIPCfg.getInstance().getCacheManager().getCache(VIPCfg.CACHE_L3);
-        if (c != null) {
-            f = c.keySet().contains(cacheKey);
-        }
-        return f;
+        return this.getCacheItem(cacheKey) != null;
     }
 
     public Map<String, String> getCacheOfStatus() {
         String cacheKey = dto.getTransStatusAsCacheKey();
-        Cache c = VIPCfg.getInstance().getCacheManager().getCache(VIPCfg.CACHE_L3);
-        if (c != null) {
-        	MessageCacheItem cacheItem = (MessageCacheItem) c.get(cacheKey);
-        	if (cacheItem != null) {
-        		return (Map<String, String>) cacheItem.getCachedData();
-        	}
+        MessageCacheItem cacheItem = (MessageCacheItem) this.getCacheItem(cacheKey);
+        if (cacheItem != null) {
+            return cacheItem.getCachedData();
         }
         return null;
     }
 
     public void addCacheOfStatus(Map<String, String> dataMap) {
         String cacheKey = dto.getTransStatusAsCacheKey();
-        Cache c = VIPCfg.getInstance().getCacheManager().getCache(VIPCfg.CACHE_L3);
-        if (c != null) {
-            c.put(cacheKey, new MessageCacheItem(dataMap));
-        }
+        addCacheItem(cacheKey, new MessageCacheItem(dataMap));
     }
 
-    public List<Locale> getSupportedLocalesFromCache() {
-        List<Locale> locales = new ArrayList<Locale>();
-        Cache c = VIPCfg.getInstance().getCacheManager().getCache(VIPCfg.CACHE_L2);
-        if (c == null) {
-            return locales;
-        }
-        Set<String> keySet = c.keySet();
-       
-        for (String key : keySet) {
-        	if (key.startsWith(ConstantsKeys.DISPNS_PREFIX)) {
-        		FormatCacheItem cacheItem = (FormatCacheItem) c.get(key);
-        		Map<String, String> langTagToDisplayNameMap = cacheItem.getCachedData();
-        		for (String languageTag : langTagToDisplayNameMap.keySet()) {
-        			locales.add(Locale.forLanguageTag(languageTag));
-        		}
-        		break;
-        	}
+    public List<Locale> getLocalesOfCachedMsgs() {
+        List<Locale> locales = new LinkedList<>();
+        Cache c = VIPCfg.getInstance().getCacheManager().getCache(VIPCfg.CACHE_L3);
+        if (c != null) {
+            Set<String> cacheKeys = c.keySet();
+            for (String key : cacheKeys) {
+                Locale locale = getLocaleByCachedKey(key);
+                if (locale != null)
+                    locales.add(locale);
+            }
         }
         return locales;
     }
 
     private Locale getLocaleByCachedKey(String key) {
+        if (key.startsWith(ConstantsKeys.DISPNS_PREFIX))
```

Comment:
We should have a prefix for translation bundles. If not and  there is a new prefix added, this line needs to be changed.

Reply:
This change is not related to this PR so I have removed it.